test/canvas-mock.js

```diff
@@ -87,4 +90,54 @@ function mockify(window) {
   };
 }
 
+
+/**
+ * Initialize the mock, jsdom and jsdom_global for unit test usage.
+ *
+ * In particular, this takes care of a 'helpful' message about support of `getContext()`, which in
+ * practice is a complete eyesore. It's not a problem in our case, because right here we're using
+ * our own canvas mock to deal with it.
+ * The unit test run fine with or without this message, it just gets in the way.
+ *
+ * @param {string} [html='']  html definitions which should be added to the jsdom definition
+ * @returns {function}  function to call in after(), to clean up for `jsdom_global`
+ */
+function mockify(html = '') {
+  // Start of message that we want to override.
+  let msg = 'Error: Not implemented: HTMLCanvasElement.prototype.getContext'
+    + ' (without installing the canvas npm package)';
+
+  // jsdom uses a virtual console to output its messages. The default virtual console is
+  // overridden here so that we can (selectively) suppress output.
+  const virtualConsole = new jsdom.VirtualConsole();
+
+  // This defines a simple 'mock' console output.
+  // Only 'error' is overridden which for current purposes is fine. The other calls we'll deal
+  // with when they crash, in the event that they get used.
+  let myConsole = {
+    error: (msg) => {
```

Comment:
Is Warning something we can also log?

Reply:
Yes, sure. But it's not used here so YAGNI until you do.

Reply:
Oh I see. I misunderstood and thought we were suppressing warnings by not including them. But now I understand that until a warning is invoked by a test, it's pointless to add it to the mock.

Reply:
mock upon mock upon mock....pretty soon it's going to be mocks all the way down.